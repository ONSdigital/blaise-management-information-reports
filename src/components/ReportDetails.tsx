import React, { ReactElement } from "react";
import { Link } from "react-router-dom";

interface ReportDetailsProps {
    link: string
    title: string
    description: string
}

function ReportDetails({
    description,
    link,
    title,
}: ReportDetailsProps): ReactElement {
    return (
        <div className="ons-grid__col ons-col-6@m">
            <div className="ons-card" aria-labelledby={link} aria-describedby={`${link}-text`}>
                <h2 className="ons-u-fs-m" id={link}>
                    <Link to={link}>
                        {title}
                    </Link>
                </h2>
                <p id="interviewer-call-history-text">{description}</p>
            </div>
        </div>
    );
}

export default ReportDetails;
