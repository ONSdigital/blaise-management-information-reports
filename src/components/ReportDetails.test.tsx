import React from "react";
import "@testing-library/jest-dom";
import { createMemoryHistory } from "history";
import { cleanup, render, waitFor } from "@testing-library/react";
import { Router } from "react-router";
import ReportDetails from "./ReportDetails";
import { act } from "react-dom/test-utils";
import flushPromises from "../tests/utilities";
import { screen } from "@testing-library/dom";

describe("ReportDetails", () => {
    it("matches snapshot", async () => {
        const history = createMemoryHistory();
        const wrapper = render(
            <Router history={history}>
                <ReportDetails link=""title="" description="" />
            </Router>
        );
        await act(async () => {
            await flushPromises();
        });

        await waitFor(() => {
            expect(wrapper).toMatchSnapshot();
        });
    });
    it("renders correctly", async () => {
        const history = createMemoryHistory();
        await act(async () => {
            render(
                <Router history={history}>
                    <ReportDetails link="blah-link" title="blah-title" description="blah-description" />
                </Router>
            );
        });
        expect(screen.queryByText("blah-title")).toBeVisible();
        expect(screen.queryByText("blah-description")).toBeVisible();
    });
    afterAll(() => {
        jest.clearAllMocks();
        cleanup();
    });
});