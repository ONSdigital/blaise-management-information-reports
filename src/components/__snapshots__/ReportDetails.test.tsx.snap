// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportDetails matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ons-grid__col ons-col-6@m"
      >
        <div
          aria-describedby="-text"
          aria-labelledby=""
          class="ons-card"
        >
          <h2
            class="ons-u-fs-m"
            id=""
          >
            <a
              href="/"
            />
          </h2>
          <p
            id="interviewer-call-history-text"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ons-grid__col ons-col-6@m"
    >
      <div
        aria-describedby="-text"
        aria-labelledby=""
        class="ons-card"
      >
        <h2
          class="ons-u-fs-m"
          id=""
        >
          <a
            href="/"
          />
        </h2>
        <p
          id="interviewer-call-history-text"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
